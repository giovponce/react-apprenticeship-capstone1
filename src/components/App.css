.darkTheme{
    background-color: #1a1a1a;
    color: #fff;

}

.lightTheme{
    background-color: #fff;
    color: #1a1a1a;
}
