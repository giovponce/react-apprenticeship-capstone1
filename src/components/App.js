import React, { useContext }from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from '../pages/Home';
import DetailsPage from '../pages/Details';
import Favorites from '../pages/Favorites';
import FavDetails from '../pages/FavDetails';
import Login from '../pages/Login';
import Header from '../components/Header';
import useAxiosFetch  from '../utils/hooks/UseAxiosFetch';
import { ThemeContext } from "../Context/ThemeContext";
import { useQuery } from '../Context/QueryContext';
// import { UserContext } from "../Context/UserContext";
import './App.css';


const API_URL = `https://www.googleapis.com/youtube/v3/search?part=snippet&type=video&maxResults=5&key=${process.env.REACT_APP_UNSPLASHED_KEY}&q=`;

function App() {
  
  const theme = useContext(ThemeContext);
  const { state }  = theme ? theme : {};
  const { darkMode } = state ? state : {};

  // const loggedUser = useContext(UserContext);

  const term = useQuery();

  const [ search, setSearch ] = React.useState('');

  const { videosList } = useAxiosFetch( 
    `${API_URL}${term}`, search
  )

  const getVideoResult = (newSearch) => {
    setSearch(newSearch)
  }

    return (
        <BrowserRouter>
          <div className={`${darkMode ? "darkTheme" : "lightTheme"}`}>
            <Header getVideoResult={getVideoResult}/>
            <Routes>
              <Route path="/"
              element={
              <Home videosList={videosList} />
              }>
              </Route>
              <Route path="/details/:id"
                element={
                  <DetailsPage 
                  videosList={videosList}
                />
                  }>
              </Route>
              <Route exact path="/favorites"
                element={
                  <Favorites />
                  }>
              </Route>
              <Route exact path="/favDetails"
                element={
                  <FavDetails />
                  }>
              </Route>
              <Route exact path="/login"
                element={
                  <Login />
                  }>
              </Route>
            </Routes>
          </div>
        </BrowserRouter>
    )
}

export default App;

