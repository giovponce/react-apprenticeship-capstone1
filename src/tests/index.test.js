import { render } from '@testing-library/react';
import App from '../Components/App';

describe('renders component', () => {
    it('renders without crashing', () => {
        render(<div>Hello</div>);
    });    
});