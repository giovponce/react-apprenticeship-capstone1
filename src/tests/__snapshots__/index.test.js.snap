// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component render app 1`] = `
<div>
  <div
    class="lightTheme"
  >
    <div
      class="sc-jrQzAO fKkZku"
    >
      <form
        class="sc-bqiRlB kmiahl"
      >
        <input
          aria-label="Search"
          class="sc-crHmcD xrrMp"
          data-testid="search-input"
          placeholder="Search"
          type="search"
          value=""
        />
        <button
          class="sc-kDTinF bzwRMz"
        >
          Search
        </button>
      </form>
      <a
        class="sc-iCfMLu jCGOcy"
        href="/"
      >
        Home
      </a>
      <a
        class="sc-iCfMLu jCGOcy"
        href="/favorites"
      >
        Favorites
      </a>
      <a
        class="sc-iCfMLu jCGOcy"
        href="/login"
      >
        Login
      </a>
      <button
        class="sc-kDTinF bzwRMz"
      >
        Change theme
      </button>
    </div>
    <div>
      <h1
        class="sc-hKwDye hCoQax"
      >
        Videos
      </h1>
      <div
        class="sc-gsDKAQ iKIdJv"
      />
    </div>
  </div>
</div>
`;
